src/main/java/org/opentripplanner/transit/model/basic/Money.java

```diff
@@ -54,4 +122,75 @@ public String localize(Locale loc) {
   public String toString() {
     return localize(Locale.ENGLISH);
   }
+
+  /**
+   * Subtract the money amount from this instance and return the result.
+   */
+  public Money minus(Money other) {
+    return op(other, o -> new Money(currency, amount - o.amount));
+  }
+
+  /**
+   * Add another money amoutn to this instance and return it.
+   */
+  public Money plus(Money other) {
+    return op(other, o -> new Money(currency, amount + o.amount));
+  }
+
+  /**
+   * Does this instance represent a larger amount than the one passed in?
+   */
+  public boolean greaterThan(Money other) {
+    return booleanOp(other, amount > other.amount);
+  }
+
+  /**
+   * Does this instance represent a smaller amount than the one passed in?
+   */
+  public boolean lessThan(Money other) {
+    return booleanOp(other, amount < other.amount);
+  }
+
+  /**
+   * The currency of this instance.
+   */
+  public Currency currency() {
+    return currency;
+  }
+
+  private boolean booleanOp(Money other, boolean amount) {
+    if (currency.equals(other.currency)) {
+      return amount;
+    } else {
+      throw new IllegalArgumentException(
+        "Cannot perform operations on %s and %s because they have unequal currency.".formatted(
+            this,
+            other
+          )
+      );
+    }
```

Comment:
This check is appears in several forms in this file and should be extracted into a method.

Reply:
aafc33683d02f415b8a4e88dcbf900d0055a07e7